Adding the new configmap detail page

codyrancher opened the pull request:
fixes #13324
part of #13325

<!-- This template is for Devs to give QA details before moving the issue To-Test -->
### Summary
Fixes #
<!-- Define findings related to the feature or bug issue. -->

### Occurred changes and/or fixed issues
<!-- Include information of the changes, including collateral areas which have been affected by this PR as requirement or for convenience. -->

### Technical notes summary
<!-- Outline technical changes which may pass unobserved or may help to understand the process of solving the issue -->

### Areas or cases that should be tested
<!-- Areas that should be tested can include Airgap checks, Rancher upgrades, K8s upgrade, etc. -->
<!-- Which browser did you use for local testing? The reviewer should test with a different browser. -->
<!-- Add missing steps or rewrite them if have been missed or to complement existing information. This should define a clear way to reproduce it and not an approximation. -->

### Areas which could experience regressions
<!-- Create a detailed list of areas to be analyzed which may be affected by the changes, which would require a prior research to avoid regressions. -->

### Screenshot/Video
<!-- Attach screenshot or video of the changes and eventual comparison if you find it necessary -->

### Checklist
- [ ] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [ ] The PR has a Milestone <!-- The milestone should automatically be assigned if the linked issue has one, but does not always happen (incorrectly linked, issue has no milestone, etc) -->
- [ ] The PR template has been filled out
- [ ] The PR has been self reviewed <!-- There are no TODOs, no incorrect files in the PR, all the required files are there, no commented out code, etc-->
- [ ] The PR has a reviewer assigned
- [ ] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [ ] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes
- [ ] The PR has been reviewed in terms of Accessibility

codyrancher commented on shell/components/Drawer/ResourceDetailDrawer/helpers.ts:
This is split up from the composables file for testing reasons. Jest doesn't seem to allow you to run tests on some methods while mocking others from the same file even though there's plenty online which states otherwise.

codyrancher commented on shell/components/form/ResourceTabs/index.vue:
```diff
@@ -233,6 +237,7 @@ export default {
     v-bind="$attrs"
     :default-tab="defaultTab"
     :resource="value"
+    :use-hash="useHash"
```
We need to be able to turn off using the hash associated with tabs used on edit pages that we display in the ResourceDetailDrawer.

When a tab is rendered or selected it updates the hash in the url. Url changes trigger the slideIn/Drawer to close. Since we shouldn't need the url to reference a tab in the drawer I think it's safe to turn it off in these circumstances.

eva-vashkevich commented on shell/components/SlideInPanelManager.vue:
```diff
@@ -15,6 +15,10 @@ const currentComponent = computed(() => store.getters['slideInPanel/component'])
 const currentProps = computed(() => store.getters['slideInPanel/componentProps']);
 
 const panelTop = computed(() => {
+  if (currentProps?.value?.top) {
```
Could you please leave this comment in the code for future reference?

codyrancher replied:
Done

eva-vashkevich commented on shell/components/Drawer/ResourceDetailDrawer/__tests__/ConfigTab.test.ts:
```diff
@@ -0,0 +1,54 @@
+    expect(component.props('name')).toStrictEqual('config-tab');
+  });
+
+  it('shoulder render a dynamic component within the .container and pass the correct props', () => {
```
typo

codyrancher replied:
I kept making this mistake and caught it quite a few times. Not sure why I continue to make the error.

Fixed in all the locations.

eva-vashkevich replied:
too much climbing

codyrancher replied:
Went climbing with @rak-phillip today lol

eva-vashkevich commented on shell/components/Resource/Detail/Additional.vue:
```diff
@@ -0,0 +1,46 @@
+<script lang="ts">
```
Isn't this only used by useDefaultWorkloadIdentifyingInformation, which isn't even used anywhere? If so, should it be a part of this change?

codyrancher replied:
You're right. There was quite a bit of workload code still in the composable. I've removed the unused code.

eva-vashkevich reviewed (changes requested):
I think this is the nicest code I've read this month :)

Some preference complaints and one real issue:
1. Styling for Labels and annotations is broken in Chrome (works in Mozilla) and   it might be related to the warning shown in the console but I doubt it.
<img width="1713" alt="Screenshot 2025-06-19 at 12 14 04 PM" src="https://github.com/user-attachments/assets/3177a58d-b972-4255-930c-e028f9d7493b" />
![Screenshot 2025-06-19 at 12 19 57 PM](https://github.com/user-attachments/assets/68655d74-b8c3-42df-bcf0-0ee15e4df191)
It's likely caused by this https://github.com/rancher/dashboard/pull/14515/files#diff-207cf7366d630a0f5f5f9bf5a359fb1273fc2a1990a37098caf0ec9eea8078c2R125
2. We might want to consider how Header looks on smaller screens and add some responsiveness. Right now it's just getting squished a little, which is common to our UI, but this might be a good chance to address it a bit (but this is more of something to consider)
<img width="795" alt="Screenshot 2025-06-19 at 3 03 40 PM" src="https://github.com/user-attachments/assets/976a5e4f-944f-44ba-8b49-9ac22432a15c" />
3. I think when user clicks on 'Show all labels' or 'Show Configuration' under Annotations we should take them to the correct tab. Which might not be possible due to the hashing change you made. This is also more of a preference.
4. Why are links in 'Show all labels' and 'Show configuration' a different color? It also kinda feels like we have too many 'Show configuration' next to each other? (but that's a preference thing)
![Screenshot 2025-06-19 at 12 42 22 PM](https://github.com/user-attachments/assets/ee9d414d-0f63-4599-aec5-5ef0222a2b5d)

codyrancher commented:
> I think this is the nicest code I've read this month :)

Thanks!

> Styling for Labels and annotations is broken in Chrome (works in Mozilla) and it might be related to the warning shown in the console but I doubt it.

You were right about the area. I decided to fix it by giving the row a width. It's not obvious why firefox and chrome render the element differently. All the same selectors are getting applied and it appears to have the same computed properties like display.

> We might want to consider how Header looks on smaller screens and add some responsiveness. Right now it's just getting squished a little, which is common to our UI, but this might be a good chance to address it a bit (but this is more of something to consider)

I think you're right but it's something we're going to need to get UX input on so I don't want it to hold this PR up. I've created https://github.com/rancher/dashboard/issues/14544 to track this and hopefully fix it before 2.12 is released. @oboc-sts is this something that you can take a look at?

>  I think when user clicks on 'Show all labels' or 'Show Configuration' under Annotations we should take them to the correct tab. Which might not be possible due to the hashing change you made. This is also more of a preference.

You're right to want this and it's something that Ovi wants. Unfortunately we just don't have a good mechanism to achieve this yet so it's not going to make it into this change/release. For tracking I have created an issue for this though https://github.com/rancher/dashboard/issues/14545

> Why are links in 'Show all labels' and 'Show configuration' a different color? It also kinda feels like we have too many 'Show configuration' next to each other? (but that's a preference thing)

They shouldn't be, it's a bug that showed up because I switched from `<router-link >` to `<a>` and forgot the `.secondary` class. Fixed.

eva-vashkevich commented:
Hi @oboc-sts, Thank you for the detailed answer! Just to confirm, "Show YAML" should be a priority if available? I am not sure if it makes sense to me, or did I misunderstand? 
@codyrancher

codyrancher commented:
@eva-vashkevich  I've updated the text for the case where labels and annotations are empty.

I'm not going to change the show yaml/show configuration link until I have more of discussion with Ovi. If that cahnges I can put that change in with a future pr.

eva-vashkevich commented:
Also @codyrancher @oboc-sts 
This is not introduced by this change, but we might want to fix it:
If we go Show configuration -> Data
Value fields look enabled but you can't actually change them and everything else is greyed out
<img width="1235" alt="Screenshot 2025-06-20 at 10 32 55 AM" src="https://github.com/user-attachments/assets/d5c947fe-73b9-4f27-85c4-3b8c2bf6c25f" />
I don't think we have to fix it as part of this PR, but we might want to revisit it if we end up doing any follow up work

eva-vashkevich reviewed (approved):
LGTM
Great work
